src/agentscope/service/reasoning/wolfram_alpha.py

```diff
@@ -0,0 +1,375 @@
+        `content`. The `status` variable is from the ServiceExecStatus enum,
+        and `content` is a dictionary containing
+        the result or error information,
+        which depends on the `status` variable.
+
+    Example:
+        .. code-block:: python
+
+            result = query_wolfram_alpha_short_answers(
+                "your_api_key",
+                "What is the capital of France?"
+            )
+            if result.status == ServiceExecStatus.SUCCESS:
+                print(result.content['result'])
+            # Output: Paris, Île-de-France, France
+    """
+    url = "http://api.wolframalpha.com/v1/result"
+    params = {"i": query, "appid": api_key}
+
+    try:
+        response = requests.get(url, params=params)
+        if response.status_code == 200:
+            return ServiceResponse(
+                status=ServiceExecStatus.SUCCESS,
+                content={"result": response.text},
+            )
+        return ServiceResponse(
+            status=ServiceExecStatus.ERROR,
+            content={
+                "error": f"HTTP Error: {response.status_code} {response.text}",
+            },
+        )
+    except Exception as e:
+        return ServiceResponse(
+            status=ServiceExecStatus.ERROR,
+            content={"error": str(e)},
+        )
+
+
+def query_wolfram_alpha_simple(
+    api_key: str,
+    query: str,
+) -> ServiceResponse:
+    """
+    Query the Wolfram Alpha Simple API. The Simple API generates full
+    Wolfram|Alpha output in a universally viewable image format.
+    Suitable for queries such as knowledge/facts retrieval. See
+    https://products.wolframalpha.com/simple-api/documentation
+    for more details.
+
+    Args:
+        api_key (`str`):
+            Your Wolfram Alpha API key.
+        query (`str`):
+            The query string to search.
+
+    Returns:
+        `ServiceResponse`: A dictionary with two variables: `status` and
+        `content`. The `status` variable is from the ServiceExecStatus enum,
+        and `content` is a dictionary containing
+        the result or error information,
+        which depends on the `status` variable.
+        The returned image is saved in the save directory as
+        `wolfram_alpha_result.png`.
+
+    Example:
+        .. code-block:: python
+
+            result = query_wolfram_alpha_simple(
+                "your_api_key",
+                "Plot sin(x)"
+            )
+            if result.status == ServiceExecStatus.SUCCESS:
+                print("Result saved as 'wolfram_alpha_result.png'")
+    """
+    url = "http://api.wolframalpha.com/v1/simple"
+    params = {"i": query, "appid": api_key}
+
+    try:
+        response = requests.get(url, params=params)
+        if response.status_code == 200:
+            with open("wolfram_alpha_result.png", "wb") as file:
+                file.write(response.content)
+            print("Result saved as 'wolfram_alpha_result.png'")
```

Comment:
It would be better to make this file path and file name configurable. (Of course, you can provide a default value for this)

Reply:
Updated according to the comment.